packages/ui-utils/src/hash.ts

```diff
@@ -64,6 +64,45 @@ function executeHash(input: string): string {
   return toBase64(String(hash))
 }
 
+// resource: https://stackoverflow.com/a/52171480
+function cyrb53(str: string, seed = 0) {
+  let h1 = 0xdeadbeef ^ seed,
+    h2 = 0x41c6ce57 ^ seed
+  for (let i = 0, ch; i < str.length; i++) {
+    ch = str.charCodeAt(i)
+    h1 = Math.imul(h1 ^ ch, 2654435761)
+    h2 = Math.imul(h2 ^ ch, 1597334677)
+  }
+  h1 =
+    Math.imul(h1 ^ (h1 >>> 16), 2246822507) ^
+    Math.imul(h2 ^ (h2 >>> 13), 3266489909)
+  h2 =
+    Math.imul(h2 ^ (h2 >>> 16), 2246822507) ^
+    Math.imul(h1 ^ (h1 >>> 13), 3266489909)
+
+  return (
+    (h2 >>> 0).toString(16).padStart(8, '0') +
+    (h1 >>> 0).toString(16).padStart(8, '0')
+  )
+}
+function hashInstance(instance: string, map: Map<string, number>) {
```

Comment:
Instead of passing a Map here, isn't it enough to pass the instance type? Then keep here a static map that keeps count of things, and this function could return the current instance number/name

Reply:
Sadly that is not enough because it will result in the client and server becoming out of sync the time a browser refresh happens.
This code will run in 2 environments in case of SSR.
1. server side
2. client side

At the beginning of everything the server and client will be sync.
Now imagine the following scenario, we have an app containing only a `Button` component:
1. server renders the component ->  server :`Button` counter: 1
2. client picks up and hydrates the app, thus rendering the same `Button` -> client: `Button` counter: 1
3. client does a whole browser refresh -> every static file (including .js) will have to be re-requested, parsed, and run and that means the `Button` counter will be again 1
4. meanwhile on the server we didn't do a restart meaning the server side counter will be: 2
5. client and server become out of sync

So in order to be sync all the time we have to reset the counter on each request on the server side, and I think this is the most elegant way to do that. (using `React.Context`)

Reply:
thanks for the explanation!